### ISSUE TITLE
Auto Execution "Always" gives flickering on Subdivide modifier

### ISSUE THREAD

**Issue by enzyme69** [JacquesLucke/animation_nodes#576]
![flicker_update](https://cloud.githubusercontent.com/assets/3952687/16973722/68a17c2e-4e79-11e6-9d9f-1063eb817991.gif)

I am using "Auto Execution" as Always. All I have in the scene is a Box with 
- Skin Modifier
- Subdiv Modifier
- Using Object Attribute Output, I am changing >> modifiers["Subsurf"].levels

I am getting Flickering mesh with Auto Execution.
When I use Property Changed tick box, the flicker disappear.

Now, I am curious on New Trigger. I will have to check that.

But I think even with "Always", it should not flickering.

**Comment by JacquesLucke** [comment]
Hm, I guess you don't change the vertices all the time? This looks a bit like a bug in the Skin Modifier. Will check in more detail when I'm at home.

**Comment by JacquesLucke** [comment]
https://developer.blender.org/T48937

**Comment by og76** [comment]
fwiw 
this is not the only place you'll encounter "different outputs for same input" behavior, more like a random choice between equivalent solutions, eventually leading to flicker in Animation Nodes or other repeated execution case.

However, **do notice** that auto execution just repeats the exec every x seconds. 
Evwn if without it things seam ok, well, they are not if u are animating. On each frame the solution will be random, so it will flicker ...

-

Another example that comes to mind is bvh closest where if the closest point is on an edge, the returned polygon index will.. flicker, being sort of a random choice between those 2 poly. 
(one of the reasons for the extra normal check I was doin on the "is inside volume - normal based" version. extra checks that eventually made it sloow.. )

There are other cases for this, (marginal or weird or not), but do not remember now 
and, sorry, time is not on my side these days. :(

ps.
solutions? hm, it so depends on the specific case. in some cases a certain tolerance (or on the contrary, precision) may help "fix" the result ... or more "fixed"

**Comment by enzyme69** [comment]
​Yes, I guess it can be a bit tidier. In fact, I still wonder why the Mesh
output from AN still cannot be "Shade Smoothed". It's always goes back to
non smooth.

**Comment by og76** [comment]
w do you mean "goes back"?

also..
![smooth](https://cloud.githubusercontent.com/assets/11709974/17081121/99e0e198-5154-11e6-985b-1fe50429a525.PNG)

**Comment by enzyme69** [comment]
Ok, I think that might have been fixed at some point I probably was using
older version. I recalled having this "issue":
1. Original Suzanne
2. Instance Copy Suzanne, Shade Smooth, and it goes to smooth and playback,
it goes back to Shade Flat.

I tried it just now, the copy when smoothed, it affects the original
Suzanne. Which is weird too. But I tested again and it is not affecting
original Suzanne. I tested with this switches on and off: Deep Copy, Copy
Full Objects, Copy from Source.

But, probably has been fixed. And thanks for pointing that "Shade Object
Smooth" node. I didn't know.

Anyhow, the flickering thing happens sometimes when I subdivide mesh
outputted from AN. It is unexpected behaviour, I reckoned.





------------------------------------------------------------
- Jimmy Gunawan
http://blendersushi.blogspot.com.au
http://mayaspiral.blogspot.com.au
http://houdoodles.blogspot.com.au
http://puppetar.blogspot.com.au
